pkg/store/filter.go

```diff
@@ -16,6 +16,10 @@ package store
 
 import topoapi "github.com/onosproject/onos-api/go/onos/topo"
 
+func matchRelation(object *topoapi.Object, filter *topoapi.RelationFilter, entities map[topoapi.ID]*topoapi.Object) bool {
+	return object.Type == topoapi.Object_RELATION && string(object.GetRelation().KindID) == filter.GetRelationKind() && string(object.GetRelation().GetSrcEntityID()) == filter.SrcId && (filter.TargetKind != "" || entities[object.GetRelation().GetTgtEntityID()].GetKind().GetName() == filter.TargetKind)
+}
```

Comment:
The last condition is filter.TargetKind != "" || entities[object.GetRelation().GetTgtEntityID()].GetKind().GetName() == filter.TargetKind). What you are trying to achieve? The idea is to ignore Target ID thing?

Reply:
The idea is if the target kind ID is specified, then we only want those kinds of objects. It's an optional secondary filter.

Reply:
Yes, if the filter target kind is not specified, then it shouldn't matter, but if it is specified the objects (relation)'s target kind should be the same as the filter kind.